src/Tutor.Infrastructure/Database/EventStore/IEventStore.cs

```diff
@@ -1,15 +1,13 @@
-﻿using System.Collections.Generic;
-using System.Threading.Tasks;
+﻿using System.Threading.Tasks;
 using Tutor.Core.BuildingBlocks;
 using Tutor.Core.BuildingBlocks.EventSourcing;
-using Tutor.Core.LearnerModel.DomainOverlay.KnowledgeComponentMasteries.Events;
 
 namespace Tutor.Infrastructure.Database.EventStore
 {
     public interface IEventStore
     {
         void Save(EventSourcedAggregateRoot aggregate);
+        IEventQueryable Events { get; }
```

Comment:
Should the repository allow this level of flexibility to its clients?
So far the repositories have encapsulated these queries.

Reply:
This level of flexibility is appropriate for IEventStore because it isn't really part of the repository layer, but rather just a generic abstraction of an event store meant to be used by specific repositories (although in this case a specific repository doesn't exist yet). It's purpose is to fill the role that EF Core has for relational databases, so it only exists because there isn't a .NET standard way to represent functionalities offered by an event store. 
Although for now it's kind of questionable if this additional layer of abstraction will actually turn out to be useful in this project or just an unnecessary complication. It mostly depends on how many different queries on IEventStore we end up needing. If it's just a few then it would be much easier to just get rid of IEventStore and rewrite all the queries directly in case of a technology swap, but if there's a lot then implementing query translation instead seems worth it.